tests/allowed_addresses.go

```diff
@@ -184,7 +184,8 @@ func AllowedAddresses(network interfaces.LocalNetwork) {
 
 	// Wait for relayer to start up
 	log.Info("Waiting for the relayer to start up")
-	testUtils.WaitForChannelClose(readyChan, 15*time.Second)
+	startupCtx, _ := context.WithTimeout(ctx, 15*time.Second)
```

Comment:
Doesn't this only give 15 seconds for us to get past line 312?

Reply:
I'm okay with setting one context for the whole test, but we should use a more generous timeout.

Reply:
Sorry, posted this on outdated code. It doesn't need to be done in this PR, but I think using `context.WithTimeout(ctx, 15*time.Second)` for every test instead of `context.Background()` and reusing the same context throughout the test would make sense.

Reply:
> Sorry, posted this on outdated code. It doesn't need to be done in this PR, but I think using `context.WithTimeout(ctx, 15*time.Second)` for every test instead of `context.Background()` and reusing the same context throughout the test would make sense.

I'm not sure I'm parsing your meaning correctly. I am calling context.WithTimeout for each call of the startup. 

You prefer using a top level context for each test that gets passed throughout? That seems more likely to cause flakes and need adjustments later on but I'm ok with that later. Are you ok with me merging this as is?

Reply:
So we call `ctx := context.Background()` on line 15, and then create a new context here. I think we should just set one context per test, and give it a reasonable timeout of 1 minute or so. Right now reusing this context in 4 places means we have to get through here to line 312 in 15 seconds, which may or may not be tight.

I'm fine with merging this as-is though